media/server/gstplayer/source/tasks/generic/SetupElement.cpp

```diff
@@ -80,11 +96,41 @@ SetupElement::~SetupElement()
 void SetupElement::execute() const
 {
     RIALTO_SERVER_LOG_DEBUG("Executing SetupElement");
-    if (m_glibWrapper->gStrHasPrefix(GST_ELEMENT_NAME(m_element), "westerossink"))
+
+    // In playbin3 AutoVideoSink uses names like videosink-actual-sink-brcmvideo whereas playbin
+    // creates sink with names brcmvideosink*, so it is better to check actual type name here
+    const std::string elementTypeName = m_glibWrapper->gTypeName(G_OBJECT_TYPE(m_element));
+
+    if (elementTypeName == "GstAutoVideoSink")
+    {
+        // Check and store child sink so we can set underlying properties
+        m_glibWrapper->gSignalConnect(m_element, "child-added", G_CALLBACK(autoVideoSinkChildAddedCallback), &m_player);
+        m_glibWrapper->gSignalConnect(m_element, "child-removed", G_CALLBACK(autoVideoSinkChildRemovedCallback),
+                                      &m_player);
+
+        // AutoVideoSink sets child before it is setup on the pipeline, so check for children here
+        GstIterator *sinks = m_gstWrapper->gstBinIterateSinks(GST_BIN(m_element));
+        if (sinks && sinks->size > 1)
+        {
+            RIALTO_SERVER_LOG_WARN("More than one child sink attached");
+        }
+
+        GValue elem = G_VALUE_INIT;
+        if (m_gstWrapper->gstIteratorNext(sinks, &elem) == GST_ITERATOR_OK)
+        {
+            m_player.addAutoVideoSinkChild(G_OBJECT(m_glibWrapper->gValueGetObject(&elem)));
+        }
+        m_glibWrapper->gValueUnset(&elem);
+
+        if (sinks)
+            m_gstWrapper->gstIteratorFree(sinks);
+    }
+
+    if (elementTypeName == "GstWesterosSink" || elementTypeName == "Gstbrcmvideosink")
```

Comment:
I think that checking if added element is type GST_ELEMENT_FACTORY_TYPE_SINK | GST_ELEMENT_FACTORY_TYPE_MEDIA_VIDEO would be more generic than checking names.

Reply:
done